jabgui/src/main/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtil.java

```diff
@@ -104,70 +216,72 @@ public LinkFilesResult linkAssociatedFiles(List<BibEntry> entries, BiConsumer<Li
-
-        LOGGER.debug("Searching for extensions {} in directories {}", extensions, directories);
-
-        // Run the search operation
-        FileFinder fileFinder = FileFinders.constructFromConfiguration(autoLinkPreferences);
-        List<Path> result = new ArrayList<>(fileFinder.findAssociatedFiles(entry, directories, extensions));
-        result.addAll(findAssociatedFilesByBrokenLinkedFile(entry));
-
-        // Collect the found files that are not yet linked
-        List<Path> linkedFiles = entry.getFiles().stream()
-                                      .map(file -> file.findIn(directories))
-                                      .filter(Optional::isPresent)
-                                      .map(Optional::get)
-                                      .toList();
-        for (Path foundFile : result) {
-            boolean fileAlreadyLinked = linkedFiles.stream()
-                                                   .anyMatch(linked -> {
-                                                       try {
-                                                           return Files.isSameFile(linked, foundFile);
-                                                       } catch (IOException e) {
-                                                           LOGGER.debug("Unable to check file identity, assuming no identity", e);
-                                                           return false;
-                                                       }
-                                                   });
-
-            if (!fileAlreadyLinked) {
-                Optional<ExternalFileType> type = FileUtil.getFileExtension(foundFile)
-                                                          .map(extension -> ExternalFileTypes.getExternalFileTypeByExt(extension, externalApplicationsPreferences))
-                                                          .orElse(Optional.of(new UnknownExternalFileType("")));
-
-                String strType = type.map(ExternalFileType::getName).orElse("");
-                Path relativeFilePath = FileUtil.relativize(foundFile, directories);
-                LinkedFile linkedFile = new LinkedFile("", relativeFilePath, strType);
-                associatedNotLinkedFiles.add(linkedFile);
-                LOGGER.debug("Found file {} for entry {}", linkedFile, entry.getCitationKey());
-            }
-        }
+        // Find the associated files
+        List<String> extensions = getConfiguredExtensions();
+        LOGGER.debug("Searching for associated not linked files with extensions {} in directories {}", extensions, directories);
+        return Stream.concat(
+                             findAssociatedNotLinkedFilesWithFinder(entry, preConfiguredFileFinder, extensions).stream(),
+                             findAssociatedNotLinkedFilesWithFinder(entry, brokenLinkedFileNameBasedFileFinder, extensions).stream())
+                     .distinct()
+                     .toList();
+    }
```

Comment:
I wish it were much clearer by writing code like this to show that we have two sources of associated files.
And the new `BrokenLinkedFileNameBasedFileFinder.java` also helped this purpose.